sor/src/main/java/com/bazaarvoice/emodb/sor/db/astyanax/CqlDataWriterDAO.java

```diff
@@ -217,16 +227,23 @@ private void deleteCompactedDeltas(ByteBuffer rowKey, WriteConsistency consisten
         // delete the old deltas & compaction records
         if (_writeToLegacyDeltaTable) {
             BatchStatement legacyTableStatement = new BatchStatement(BatchStatement.Type.UNLOGGED);
-            for (UUID change : changesToDelete) {
-                legacyTableStatement.add(deleteStatement(placement.getDeltaTableDDL(), rowKey, change, consistencyLevel));
+            for (DeltaClusteringKey change : changesToDelete) {
+                legacyTableStatement.add(deleteStatement(placement.getDeltaTableDDL(), rowKey, change.getChangeId(), consistencyLevel));
             }
             legacyTableFuture = session.executeAsync(legacyTableStatement);
         }
 
         if (_writeToBlockedDeltaTable) {
             BatchStatement newTableStatement = new BatchStatement(BatchStatement.Type.UNLOGGED);
-            for (UUID change : changesToDelete) {
-                newTableStatement.add(deleteStatement(placement.getBlockedDeltaTableDDL(), rowKey, change, consistencyLevel));
+            for (DeltaClusteringKey change : changesToDelete) {
+                if (change.hasNumBlocks() && change.getNumBlocks() < 2) {
```

Comment:
blocks(0, 1) = 2 NumBlocks. So may be "change.getNumBlocks() <= 2"  instead of "change.getNumBlocks() < 2"? Am I missing something :?

Reply:
Agreed, good catch!